src/main/java/com/github/minecraftschurlimods/arsmagicalegacy/compat/jei/SkillIngredient.java

```diff
@@ -0,0 +1,72 @@
+package com.github.minecraftschurlimods.arsmagicalegacy.compat.jei;
+
+import com.github.minecraftschurlimods.arsmagicalegacy.api.skill.Skill;
+import com.github.minecraftschurlimods.arsmagicalegacy.client.ClientHelper;
+import com.github.minecraftschurlimods.arsmagicalegacy.client.SkillIconAtlas;
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.PoseStack;
+import mezz.jei.api.ingredients.IIngredientHelper;
+import mezz.jei.api.ingredients.IIngredientRenderer;
+import mezz.jei.api.ingredients.IIngredientType;
+import mezz.jei.api.ingredients.subtypes.UidContext;
+import net.minecraft.client.gui.GuiComponent;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.TooltipFlag;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+public class SkillIngredient implements IIngredientType<Skill> {
+    public static final IIngredientType<Skill> TYPE = () -> Skill.class;
+
+    @Override
+    public Class<? extends Skill> getIngredientClass() {
+        return Skill.class;
+    }
```

Comment:
this seems wrong

Reply:
What exactly? I could of course change `getIngredientClass()` to return `TYPE.getIngredientClass()` instead of returning `Skill.class` directly, but the result would be the same. `IIngredientType` is a functional interface, so that's fine too.

Reply:
why is SkillIngredient implementing the interface when you are using the functional interface anyway

Reply:
TYPE should either be the singleton instance of SkillIngredient or SkillIngredient should not implement the interface